add pedagogy Intro and Syllabus (#48)
data-8/zero-to-data-8: content/v2/pedagogy/syllabus.md

# Syllabus 

This section covers how to build your course syllabus and materials. It also goes over 
a few tips on how to distribute your content for Data 8.

> course.

Reviewer: ```suggestion
```

## Course syllabus page

You may use whatever technology you prefer for managing your course and
distributing content. However, we recommend setting up a syllabus page that
is used for distributing interact links and course materials. For an example,
see the structure of the Spring 2018 course syllabus:

> [http://data8.org/sp18/]

Reviewer: Does this link work? Have you tested it yet in a built page?

The syllabus has the following structure:

| Date          | Topic | Lecture  | Reading | Assignment
| ------------- | ----- | -------  | ------- | ----------
| 2018-01-09 | Cause and Effect | [Slides](http://data8.org/materials-sp18/lec/lec02PDF.pdf), [Video](https://www.youtube.com/watch?v=oF9rA2PYG6A) | [Chapter 2](https://www.inferentialthinking.com/chapters/02/causality-and-experiments.html) | [Homework 01](http://datahub.berkeley.edu/hub/user-redirect/git-sync?repo=https://github.com/data-8/materials-sp18&subPath=materials/sp18/hw/hw01/hw01.ipynb)
| 2018-01-22 | Tables | [Demos](http://datahub.berkeley.edu/hub/user-redirect/git-sync?repo=https://github.com/data-8/materials-sp18&subPath=lec/lec03.ipynb), [Slides](http://data8.org/materials-sp18/lec/lec03PDF.pdf), [Video](https://www.youtube.com/watch?v=NLw4egmXBHM) | [Chapter 3](https://www.inferentialthinking.com/chapters/03/programming-in-python.html)| |

> | 2018-01-24 | Data Types | [Demos](http://datahub.berkeley.edu/hub/user-redirect/git-sync?repo=https://github.com/data-8/materials-sp18&subPath=lec/lec04.ipynb), [Slides](http://data8.org/materials-sp18/lec/lec04PDF.pdf), [Video](https://www.youtube.com/watch?v=O3AAJs7dx-c) | [Chapters 4](https://www.inferentialthinking.com/chapters/04/data-types.html), [5](https://www.inferentialthinking.com/chapters/05/sequences.html)| [Lab 02: Types and Sequences](http://datahub.berkeley.edu/hub/user-redirect/git-sync?repo=https://github.com/data-8/materials-sp18&subPath=materials/sp18/lab/lab02/lab02.ipynb)

Reviewer: Make a footnote here that the videos and materials seen here are restricted to berkeley.edu addresses. then, redirect them to here: https://github.com/data-8/materials-fa19

Each row is a lecture, and each column is a type of material you can distribute.

> The links in the columns either point to pages on the course textbook (at inferentialthinking.com)

Reviewer: ```suggestion
The links in the columns either point to pages on the [course textbook](https://inferentialthinking.com).
```

or interact links that connect students with the course JupyterHub for distribution of homeworks and labs.

## Interact links

The primary way that material is distributed in Data 8 is via "interact links". These are links that,

> when clicked, will automatically pull new material into a students' file system on your JupyterHub, and then

Reviewer: ```suggestion
when clicked, will automatically pull new material into a student's file system on your (the instructor's) JupyterHub, and then
```

direct them to this new content in a live session.

The tool that you use to do this is called [nbgitpuller](https://github.com/data-8/nbgitpuller),
which is pre-installed in the Docker image for the course. This means you can create interact links
without needing to install anything new. For example, the following link should pull in
the [Data Science Handbook](https://github.com/jakevdp/PythonDataScienceHandbook/tree/master/notebooks) from Jake van der Plas:

`http://<< YOUR-HUB-IP >>/hub/user-redirect/git-pull?repo=https://github.com/jakevdp/PythonDataScienceHandbook&branch=master&subPath=notebooks`

In the following sections we'll cover how interact links can be used with each type of Data 8
content in order to distribute materials to your students.

## Homework

Homework in Data 8 is used for course grading as well as to guide students through the

> material. The homework uses a package called OKpy for auto-grading student homework. This can be

Reviewer: ```suggestion
material. The homework uses a package called OkPy for auto-grading student homework. This can be
```

Reviewer: Add a link as well to OkPy

used in two primary ways:

1. Grading is performed locally (on the students own computer) in order to give live feedback
   to the code that students write.

> 2. Grading is performed with the OKpy servers. This requires an account with OKpy, and is

Reviewer: ```suggestion
2. Grading is performed with the OkPy servers. This requires an account with OkPy, and is
```

   used by Data 8 to handle "official" grades for the course.

At the moment, we recommend using only #1 for your course.

The latest links to the homeworks can be found in the course page for the current

> semester. For example, the Spring 2018 course is found here:

Reviewer: ```suggestion
semester. For example, the Fall 2019 course is found here:
```

> [http://data8.org/sp18/]

Reviewer: ```suggestion
[http://data8.org/fa19/]
```

These materials are freely available on the semester course repository.
There is a new repository created for each semester of the course, as the materials
tend to evolve over time. For example, here is the repository for the Spring 2018 course:

> [https://github.com/data-8/materials-sp18]

Reviewer: ```suggestion
[https://github.com/data-8/materials-fa19]
```

you can find the homeworks for this repository here:

> [https://github.com/data-8/materials-sp18/tree/master/materials/sp18/hw]

Reviewer: ```suggestion
[https://github.com/data-8/materials-fa19/tree/master/materials/fa19/hw]
```

### Grading course homework

> Grading in Data 8 is performed with the [OKpy package](https://okpy.org/). This is a course

Reviewer: ```suggestion
Grading in Data 8 is performed with the [OkPy package](https://okpy.org/). This is a course
```

management tool that allows you to both collect and grade notebooks, take
attendance, and manage many other course tasks.

> If you prefer a more light-weight solution, we recommend looking into [Gofer Grader](https://github.com/data-8/Gofer-Grader).

Reviewer: ```suggestion
If you prefer a more lightweight solution, we recommend looking into [Gofer Grader](https://github.com/data-8/Gofer-Grader).
```

This package focuses on auto-grading Jupyter Notebooks only, and is less complex to run.

Gofer Grader works with the same API that `OKpy` uses, meaning that using one package
or another should require no change in the code itself. The only difference is that
Gofer Grader will not send any course material to the OKpy servers.

> **Note:** `GoferGrader` and `OKpy` should not both be installed at the same time. If one

>   is installed, the other should be uninstalled.

Reviewer: Gofer Grader is no longer supported and is deprecated. Let's push more for Otter Grader language, and link back to the corresponding pages in the tech guide

### Customizing your own copies of the homework

To customize the Data 8 homework for your course, take the following steps:

> 1. Fork and clone the Data 8 materials for the semester of your choice (see above section on labs).

Reviewer: ```suggestion
1. Fork and clone the Data 8 materials from the [course's Github](https://github.com/data-8/) for the semester of your choice. We recommend cloning specifically from the [Fall 2019 repository](https://github.com/data-8/materials-fa19). For more information about cloning and forking, please visit the guide on how to do so [here](https://help.github.com/en/github/getting-started-with-github/fork-a-repo).
```

> 2. Ensure that you have a JupyterHub running the Data 8 environment (if not, [follow these instructions](customize_hub_environment.html))

Reviewer: Does this link work?

3. Create your interact links so they point to *your* version of the homeworks. For example,

       http://<< YOUR-HUB-URL >>/hub/user-redirect/git-sync?repo=https://github.com/<< YOUR-USERNAME >>/materials-sp18/materials/sp18/hw/hw07/hw07.ipynb

## Labs

Alongside the textbook are several computational labs that let students interact with the
ideas covered in class. They can all be run interactively in the Data 8 environment (the same
environment that we've used in your JupyterHub).

The latest links to the labs can be found in the course page for the current

> semester. For example, the Spring 2018 course is found here:

Reviewer: ```suggestion
semester. For example, the Fall 2019 course is found here:
```

> [http://data8.org/sp18/]

Reviewer: ```suggestion
[http://data8.org/fa19/]
```

These materials are freely available on the semester course repository.
There is a new repository created for each semester of the course, as the materials

> tend to evolve over time. For example, here is the repository for the Spring 2018 course:

Reviewer: ```suggestion
tend to evolve over time. For example, here is the repository for the Fall 2019 course:
```

> [https://github.com/data-8/materials-sp18]

Reviewer: ```suggestion
[https://github.com/data-8/materials-fa19]
```

you can find the labs for this repository here:

> [https://github.com/data-8/materials-sp18/tree/master/materials/sp18/lab]

Reviewer: ```suggestion
[https://github.com/data-8/materials-fa19/tree/master/materials/fa19/lab]
```

### Customizing your own copies of the labs

All labs should already be runnable on the environment that the Data 8 Docker image serves.
The Data 8 materials are shared by adding
"interact links" to the following syllabus page:

[http://data8.org/sp18/]

We recommend forking and customizing your own version of the labs, so that you can ensure the
content is the right match for your course. To do this, perform the following steps:

1. Fork the repository for the course you want to run. (e.g., `https://github.com/data-8/materials-sp18/`)
2. Clone your repository to your machine

       git clone https://github.com/<YOUR-USERNAME>/materials-sp18/

3. Make modifications to the labs as you see fit, push them to GitHub
4. Ensure that you have a JupyterHub running the Data 8 environment (if not, [follow these instructions](customize_hub_environment))
5. Create your interact links so they point to *your* version of the labs. For example,

       http://<< YOUR-HUB-URL >>/hub/user-redirect/git-sync?repo=https://github.com/<< YOUR-ORG >>/materials-sp18&subPath=materials/sp18/lab/lab02/lab02.ipynb

> 

Reviewer: Do you think we need this section? It's almost identical to the homeworks section.
